engine/server/engine/server/websocket_api_handler.go

```diff
@@ -245,32 +274,43 @@ func streamStarlarkLogsWithHTTP[T any](ctx echo.Context, streamerPool streaming.
 	}
 }
 
-func streamServiceLogsWithWebsocket(ctx echo.Context, streamer streaming.ServiceLogStreamer) {
-	websocket.Handler(func(ws *websocket.Conn) {
-		defer ws.Close()
-		err := streamer.Consume(func(logline *api_type.ServiceLogs) error {
-			err := websocket.JSON.Send(ws, logline)
-			if err != nil {
-				return err
-			}
-			return nil
+func streamServiceLogsWithWebsocket(ctx echo.Context, cors cors.Cors, streamer streaming.ServiceLogStreamer) {
+	wsPump, err := NewWebsocketPump[api_type.ServiceLogs](ctx, cors)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to start websocket connection")
```

Comment:
I think you could create a helper function here to dry a bit, it's the same used in `streamStarlarkLogsWithWebsocket`